File: core/src/main/java/com/sequenceiq/cloudbreak/controller/mapper/HttpMediaTypeNotSupportedExceptionMapper.java
package com.sequenceiq.cloudbreak.controller.mapper;

import javax.ws.rs.core.Response.Status;

import org.springframework.stereotype.Component;
import org.springframework.web.HttpMediaTypeNotSupportedException;

import com.sequenceiq.cloudbreak.exception.mapper.BaseExceptionMapper;

@Component
public class HttpMediaTypeNotSupportedExceptionMapper extends BaseExceptionMapper<HttpMediaTypeNotSupportedException> {

    @Override
    public Status getResponseStatus(HttpMediaTypeNotSupportedException exception) {
        return Status.NOT_ACCEPTABLE;
    }

    @Override
    public Class<HttpMediaTypeNotSupportedException> getExceptionType() {
        return HttpMediaTypeNotSupportedException.class;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/mapper/HttpMediaTypeNotSupportedExceptionMapper.kt
package com.sequenceiq.cloudbreak.controller.mapper

import javax.ws.rs.core.Response
import javax.ws.rs.ext.ExceptionMapper
import javax.ws.rs.ext.Provider

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.HttpMediaTypeNotSupportedException

import com.sequenceiq.cloudbreak.logger.MDCBuilder

@Provider
class HttpMediaTypeNotSupportedExceptionMapper : ExceptionMapper<HttpMediaTypeNotSupportedException> {

    override fun toResponse(exception: HttpMediaTypeNotSupportedException): Response {
        MDCBuilder.buildMdcContext()
        LOGGER.error(exception.message, exception)
        return Response.status(Response.Status.NOT_ACCEPTABLE).entity(exception.message).build()
    }

    companion object {

        private val LOGGER = LoggerFactory.getLogger(HttpMediaTypeNotSupportedExceptionMapper::class.java)
    }
}
